source/common/upstream/cluster_manager_impl.h

```diff
@@ -113,11 +113,13 @@ class ClusterManagerInitHelper : Logger::Loggable<Logger::Id::upstream> {
     // Initial state. During this state all static clusters are loaded. Any phase 1 clusters
     // are immediately initialized.
     Loading,
-    // During this state we wait for all static clusters to fully initialize. This requires
-    // completing phase 1 clusters, initializing phase 2 clusters, and then waiting for them.
-    WaitingForStaticInitialize,
-    // If CDS is configured, this state tracks waiting for the first CDS response to populate
-    // clusters.
+    // During this state we wait to start initializing secondary clusters. In this state all
+    // phase 1 clusters have completed initialization. Initialization of the secondary clusters
+    // is started by the `initializeSecondaryClusters` method.
+    WaitingForSecondaryInitialize,
```

Comment:
I'd be a fan of adding Rtds as a specific state.

Reply:
From my neophyte perspective this would break abstraction, i.e. why should cluster manager be concerned with RTDS and reflect it in its internal state? The way I wanted to code this is:

1. Initialize primary clusters. 
1. Let the server do something else. (cluster manager is in the `WaitingForSecondaryInitialize` state).
1. Initialize secondary clusters when told so by the server.

Reply:
I think the way you have it now is clean, without any mention of RTDS inside ClusterManager, resolved.

Reply:
Thoughts on changing this to `WaitingToStartSecondaryInitialization`? I found this confusing on read through (not that what was there before was not confusing). Feel free to update others to make them more clear if that can be done. Perhaps `WaitingToStartCdsInitialization`, etc.?

Reply:
Renamed states. Cleaned-up comments a bit as well.